src/databricks/labs/ucx/source_code/dependency_resolvers.py

```diff
@@ -93,3 +131,104 @@ def _is_whitelisted(self, name: str) -> bool:
                 )
             )
         return True
+
+
+class SitePackagesResolver(BaseDependencyResolver):
+
+    def __init__(self, site_packages: SitePackages, file_loader: LocalFileLoader, syspath_provider: SysPathProvider):
+        super().__init__()
+        self._site_packages = site_packages
+        self._file_loader = file_loader
+        self._syspath_provider = syspath_provider
+
+    def resolve_import(self, name: str, problem_collector: Callable[[DependencyProblem], None]) -> Dependency | None:
+        site_package = self._site_packages[name]
+        if site_package is not None:
+            container = SitePackageContainer(self._file_loader, site_package, self._syspath_provider)
+            return Dependency(WrappingLoader(container), Path(name))
+        return super().resolve_import(name, problem_collector)
+
+
+class DependencyResolver:
+
+    @staticmethod
```

Comment:
```suggestion
    @classmethod
```
following the patterns of `from_dict(...)` in the SDK

Reply:
done